### PR TITLE
CE-79 MongoDB Plugin Docs : Addition of description of expected backup files format

### PR DESCRIPTION
**Pull request by chetanchoudhary-delphix**
1. Offlinemongodump specific validation
When dsource type is selected to offlinemongodump, make sure that following parameters are not empty.
Backup Metadata file name
Backup directory for mongodump (Also check if its a directory)

2. Sharded specific validation
When dsource type is selected to shardedsource, make sure that following parameters are not empty.
If Cluster to Cluster Sync is not selected,
Backup Metadata File name
Config Server Backup File (Also check if it's a tar / tar.gz file)
Shard Backup Files (Also check if it's a tar / tar.gz file)

3. When dsource type is selected to nonshardedsource, make sure that following parameters are not empty.
Backup Metadata File name
Config Server Backup File (Also check if it's a tar / tar.gz file)

### CODE REVIEW

**Review by manaskhandeshe-delphix — APPROVED** [review]

**Review by manaskhandeshe-delphix — DISMISSED** [review]

**Review by manaskhandeshe-delphix — CHANGES_REQUESTED** [review]

File: src/operations/common.py [hunk 1]
```diff
@@ -3815,6 +3815,55 @@ def _validate_clustersync_user_inputs(source_obj) -> None:
...
+    configserver_backup_exists = os_lib_obj.check_file_dir_exists(
+        path=backup_file,
+        is_directory=False,
+    )
+    if not configserver_backup_exists:
+        raise UserError(
+            "Config server backup file should be present on host. Given backup file not available. "
+        )
+    else:
+        is_supported = backup_file.endswith(
```
Comment (manaskhandeshe-delphix) [inline src/operations/common.py:3838]: We could create a generic function inside os_lib - check_file_type() which implements this logic and checek_ops_center_backup_files() should call that function.

File: src/operations/common.py [hunk 2]
```diff
@@ -3824,125 +3873,62 @@ def check_input_parameters(source_obj, is_dsource=False):
...
+                        check_ops_center_backup_files(
+                            backup_file=shard_backup,
+                            os_lib_obj=source_obj.os_lib_obj,
                         )
-                    )
-                    if not is_supported:
+                    except UserError as exc:
                         raise UserError(
-                            "Replicaset backup file should be of type .tar / .tar.gz format. "
+                            f"Shard backup file: {shard_backup} should be present on host. "
```
Comment (manaskhandeshe-delphix) [inline src/operations/common.py:3930]: check_ops_center_backup_files() throws one out of two exceptions.

1. "Config server backup file should be present on host. Given backup file not available. "
2. "Config server backup file should be of type .tar / .tar.gz format. "


So exc will already have the exact error message. We should display that message except _either or_.

**Review by manaskhandeshe-delphix — CHANGES_REQUESTED** [review]

**Review by manaskhandeshe-delphix — APPROVED** [review]

**Comment by manaskhandeshe-delphix** [comment]
Code looks okay. Before merge, let's hear more from @ayesharustgi-delphix

File: src/operations/common.py [hunk 3]
```diff
@@ -3193,46 +3824,170 @@ def _validate_clustersync_user_inputs(source_obj) -> None:
...
+                    f"in case of DSource type: {source_obj.parameters.d_source_type}. "
+                )
+
+        if source_obj.parameters.d_source_type == "offlinemongodump":
+            # Check if input provided for mongdump backup directory is a directory
+            try:
+                check_mongodump_utility_dir(
+                    backup_dir=source_obj.parameters.config_backupfile,
+                    os_lib_obj=source_obj.os_lib_obj,
+                )
```
Comment (ayesharustgi-delphix) [inline src/operations/common.py:3902]: why are we raising error  twice? Once inside `check_mongodump_utility_dir` and once outside. 
Either we can change return type of to function to boolean or we can use the error we raised inside.

File: src/operations/common.py [hunk 4]
```diff
@@ -3193,46 +3824,170 @@ def _validate_clustersync_user_inputs(source_obj) -> None:
...
+
+        if (
+            source_obj.parameters.d_source_type == "shardedsource"
+            or source_obj.parameters.d_source_type == "nonshardedsource"
+        ) and not source_obj.parameters.enable_clustersync:
+            try:
+                check_ops_center_backup_files(
+                    backup_file=source_obj.parameters.config_backupfile,
+                    os_lib_obj=source_obj.os_lib_obj,
+                )
```
Comment (ayesharustgi-delphix) [inline src/operations/common.py:3916]: same here

File: src/operations/common.py [hunk 5]
```diff
@@ -3193,46 +3824,170 @@ def _validate_clustersync_user_inputs(source_obj) -> None:
...
+                    "in case of DSource type Sharded source. "
+                )
+            for shard_backup_dict in source_obj.parameters.shard_backupfiles:
+                shard_backup = shard_backup_dict.get("backupfile")
+                if shard_backup:
+                    try:
+                        check_ops_center_backup_files(
+                            backup_file=shard_backup,
+                            os_lib_obj=source_obj.os_lib_obj,
+                        )
```
Comment (ayesharustgi-delphix) [inline src/operations/common.py:3936]: same here too

File: src/operations/common.py (hunk 3 above)
Comment (chetanchoudhary-delphix) [reply src/operations/common.py]: Updating this by adding the raise_exception flag in the function.

**Pull request merged**